### PR TITLE
Fix NormalizeL2Fusion and allow LpNormalization to be fused to Normal…

### PR DESCRIPTION
**Pull request by mateusztabaka**
…izeL2

Ticket: 54937

- perform NormalizeL2Fusion on CPU only if axes = [1] or [1, 2, ...]
- fix NormalizeL2FusionWithMax and NormalizeL2FusionWithAdd patterns:
    use x / sqrt(eps_mode(reduce(x**2), eps)) instead of x / eps_mode(sqrt(reduce(x**2)), eps)
- use Power instead of Multiply in builder::opset1::l2_norm
- add keep_dims argument to builder::opset1::lp_norm and use it when handling LpNormalization in ONNX importer (so we can remove Broadcast there).

### CODE REVIEW

**Review by postrational — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp [hunk 1]
```diff
@@ -277,6 +278,23 @@ static void Transformation(CNNNetwork& clonedNetwork, const Config& conf) {
                 return node->input_value(0).get_partial_shape().rank().get_length() > 5;
             });
 
+    auto normalizeL2FusionCallback = [](const_node_ptr &node) -> bool {
+        auto axes_node = std::dynamic_pointer_cast<const ngraph::opset4::Constant>(node);
+        if (!axes_node)
+            return true;
+        auto axes = axes_node->cast_vector<size_t>();
```
Comment (GlebKazantaev) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:285]: We usually do not use size_t to get values from constant due to its variety on platforms. 
```suggestion
        auto axes = axes_node->cast_vector<uin64_t>();
```

Comment (GlebKazantaev) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:285]: Please add comment with explanations. Why is it required by CPU plugin? is it temporary limitation and so on.

File: inference-engine/src/transformations/src/transformations/common_optimizations/normalize_l2_fusion.cpp [hunk 2]
```diff
@@ -25,14 +25,17 @@ ngraph::pass::NormalizeL2FusionWithMax::NormalizeL2FusionWithMax() {
     auto pow = std::make_shared<ngraph::opset4::Power>(input, exp);
     auto axes = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
     auto reduce_sum = std::make_shared<ngraph::opset4::ReduceSum>(pow, axes);
-    auto sqrt = std::make_shared<ngraph::opset4::Sqrt>(reduce_sum);
     auto eps_const = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
-    auto sqrt_max_eps = std::make_shared<ngraph::opset4::Maximum>(sqrt, eps_const);
-    auto divide = std::make_shared<ngraph::opset4::Divide>(input, sqrt_max_eps);
+    auto max = std::make_shared<ngraph::opset4::Maximum>(reduce_sum, eps_const);
+    auto sqrt = std::make_shared<ngraph::opset4::Sqrt>(max);
+    auto divide = std::make_shared<ngraph::opset4::Divide>(input, sqrt);
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/normalize_l2_fusion.cpp:31]: Update or add new tests.

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/normalize_l2_fusion.cpp (hunk 2 above)
Comment (mateusztabaka) [reply inference-engine/src/transformations/src/transformations/common_optimizations/normalize_l2_fusion.cpp:31]: I've updated the tests here inference-engine/tests/functional/inference_engine/transformations/normalize_l2_fusion_test.cpp

**Review by GlebKazantaev — APPROVED** [review]

**Comment by GlebKazantaev** [comment]
@dmitry-gorokhov  do we need to run internal validation to verify this changes?

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp [hunk 3]
```diff
@@ -277,6 +278,25 @@ static void Transformation(CNNNetwork& clonedNetwork, const Config& conf) {
                 return node->input_value(0).get_partial_shape().rank().get_length() > 5;
             });
 
+    auto normalizeL2FusionCallback = [](const_node_ptr &node) -> bool {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:281]: Actually `isSupportedOperation` was designed to provide an ability to check if the ngraph operation is supported by the plug-in or not. Besides other usecases the intention was to simplify transformation pipeline adjustments (by reusing  `isSupportedOperation` inside calls callbacks). 
In current Normalize fusion implementations `transformation_callback` is called with constant node (not resulting normalize) which makes impossible reuse of `isSupportedOperation`.
So my question is: does is makes sense to change fusing transformations design in order to call callback with resulting node? I understand there might be negative impacts, but still want to discuss the topic from general design perspective.

cc'ed @GlebKazantaev

**Comment by dmitry-gorokhov** [comment]
> @dmitry-gorokhov do we need to run internal validation to verify this changes?

For sure. I will trigger it

**Comment by dmitry-gorokhov** [comment]
> > @dmitry-gorokhov do we need to run internal validation to verify this changes?
> 
> For sure. I will trigger it

Internal validation passes. Results look good

**Review by dmitry-gorokhov — APPROVED** [review]
CPU part lgtm

**Review by ilyachur — APPROVED** [review]

**Pull request merged**